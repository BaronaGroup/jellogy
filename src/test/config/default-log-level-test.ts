import {prepareTestKarhu} from '../test-utils'

describe('default-log-level-test', () => {
  const karhuTest = prepareTestKarhu({
    defaultLogLevel: 'WARN'
  })
  it('allows filtering messages that do not meet a priority threshold (unless overridden)', karhuTest((karhu, output) => {
    const ctx = karhu.context('test-context')
    ctx.error('Should show up')
    ctx.warn('Should also show up')
    ctx.info('Nope')
    ctx.debug('Not this either')
    expect(output.tracked).toMatchSnapshot()
  }))
})
