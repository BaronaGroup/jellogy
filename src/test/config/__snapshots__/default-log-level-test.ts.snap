// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-log-level-test allows filtering messages that do not meet a priority threshold (unless overridden) 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [
    Array [
      "[2] [93mWARN test-context Should also show up[39m",
      "WARN",
      "test-context",
    ],
  ],
  "ERROR" => Array [
    Array [
      "[1] [97m[101mERROR test-context Should show up[49m[39m",
      "ERROR",
      "test-context",
    ],
  ],
}
`;
